Added to plugin optional opportunity generation Play client. (#366)
playframework/play-soap: docs/modules/ROOT/pages/plugin/how-to-use.adoc

`scala` - `scala.concurrent.Future` for Scala projects.

`java` - `java.util.concurrent.CompletionStage` for Java projects.

> |-xjc-Xplay:target
> |play |false |Generate the client for SOAP service.
> 
> `play` - client for Play

Reviewer: ```suggestion
|-xjc-Xplay:target
|play |false |Generate the additional useful classes to using SOAP service with specified framework.

`play` - Play Framework
```

Reviewer: @BillyAutrey Hi, Billy! 👋 Can you help me with this formulation? Is my suggestion ok?

Reviewer: Hi, @BillyAutrey!

> Is the idea here that specifying the `play` parameter will generate Play Framework client code?

Yes, you are right.

> Is the code truly a client, or a service?

The name of this class has postfix `Service`. But this class helps to developer create a client for external SOAP service 😄 For example:

```scala
@javax.inject.Singleton
class PrimitivesService @javax.inject.Inject() (apacheCxfBus: play.soap.ApacheCxfBus, configuration: play.api.Configuration) extends play.soap.PlaySoapClient(apacheCxfBus, configuration) {

  lazy val primitives: Primitives = {
    createPort[Primitives](new QName("http://testservice.soap.play/primitives"), "Primitives", "http://0.0.0.0:8080/primitives")
  }

  def primitives(handlers: Handler[_ <: MessageContext]*): Primitives = {
    createPort[Primitives](new QName("http://testservice.soap.play/primitives"), "Primitives", "http://0.0.0.0:8080/primitives", handlers: _*)
  }

  @annotation.varargs
  def getPrimitives(handlers: Handler[_ <: MessageContext]*): Primitives = primitives(handlers: _*)
}
```

> Is there intent to extend this for additional frameworks/clients?

Yes, my long-term plan has this extending. That why I requested @atyutin90 to add special parameter for that. At least, if someone wants to use Play SOAP library without Play Framework, he will can disable generation of classes that has an import of Play Framework classes.

Reviewer: > The options are also a little weird here. For Scala readability, I might make the options for `target` either `play` or `none`.

If you about `false`, it isn't a value of `target` param, it's a value of _Required_ column.
![2023-05-19_18-05-37 (2)](https://github.com/playframework/play-soap/assets/817328/59272e4b-1672-4714-9468-fed1cda20aee)

> |play |false |Generate the additional useful classes to using SOAP service with specified framework.

Reviewer 2: ```suggestion
|play |none |Generates SOAP client classes for the specified framework.
```

Reviewer: I changed the text in the last column by your suggestion. But I kept `false` value for `Required` column. WDYT,  @BillyAutrey?

Reviewer 2: LGTM!
